frontend/src/scenes/plugins/plugin/pluginLogsLogic.ts

```diff
@@ -3,13 +3,35 @@ import api from '~/lib/api'
 import { PluginLogEntry } from '~/types'
 import { teamLogic } from '../../teamLogic'
 import { pluginLogsLogicType } from './pluginLogsLogicType'
+import { CheckboxValueType } from 'antd/lib/checkbox/Group'
 
 export interface PluginLogsProps {
     pluginConfigId: number
 }
 
 export const LOGS_PORTION_LIMIT = 50
 
+const makeLogsAPICall = async (
+    pluginConfigId: number,
+    currentTeamId: number | null,
+    searchTerm: string,
+    typeFilters: CheckboxValueType[],
+    trailingEntry: PluginLogEntry | null = null,
+    leadingEntry: PluginLogEntry | null = null
+): Promise<PluginLogEntry[]> => {
+    const type_filters =
+        typeFilters && typeFilters.length > 0 ? `&type_filter=${typeFilters.join('&type_filter=')}` : ''
+    const search = searchTerm ? `&search=${searchTerm}` : ''
+    const before = trailingEntry ? '&before=' + trailingEntry.timestamp : ''
+    const after = leadingEntry ? '&after=' + leadingEntry.timestamp : ''
+
+    const response = await api.get(
+        `api/projects/${currentTeamId}/plugin_configs/${pluginConfigId}/logs?limit=${LOGS_PORTION_LIMIT}${before}${after}${search}${type_filters}`
+    )
+
+    return response.results
```

Comment:
There's now a API composition/building approach inside `lib/api.ts`. Perhaps `api.pluginLogs.search()` is a better place for this code?